Magenic.BadgeApplication/Magenic.BadgeApplication.DataAccess.EF/SendMessageDAL.cs

```diff
@@ -51,49 +68,59 @@ public void SendActivityNotifications()
         {
             using (Entities context = new Entities())
             {
-                var activitySubmissions = context.ActivitySubmissions
-                    .Where(sub => sub.ItemStatu.ItemStatusId == (int)ActivitySubmissionStatus.AwaitingApproval)
-                    .ToList();
+	            IList<Employee> peopleToEmail = getPeopleToEmail( context );
 
-                var employees = context.Employees.ToList();
-                var peopleToEmail = activitySubmissions
-                    .Join(employees, asub => asub.EmployeeId, emp => emp.EmployeeId, (asub, emp) => new { asub = asub, emp = emp });
+	            if ( 0 == peopleToEmail.Count )
+		            return;
 
-                var emailAddresses = new List<string>();
-                foreach (var person in peopleToEmail)
-                {
-                    if (person.emp.ApprovingManagerId1.HasValue)
-                    {
-                        var manager = employees
-                            .Where(emp => emp.EmployeeId == person.emp.ApprovingManagerId1.Value)
-                            .SingleOrDefault();
-
-                        if (manager != null)
-                        {
-                            emailAddresses.Add(manager.EmailAddress);
-                        }
-                    }
-
-                    if (person.emp.ApprovingManagerId2.HasValue)
-                    {
-                        var manager = employees
-                            .Where(emp => emp.EmployeeId == person.emp.ApprovingManagerId2.Value)
-                            .SingleOrDefault();
-
-                        if (manager != null)
-                        {
-                            emailAddresses.Add(manager.EmailAddress);
-                        }
-                    }
-                }
+				IList<Employee> employees = getEmployees( context );
+
+	            IList<string> emailAddresses = getEmailAddresses( peopleToEmail, employees );
 
-                // TODO: figure out a better way to do this. Maybe using RazorEngine?
-                emailAddresses = emailAddresses.Distinct().ToList();
                 var emailSubject = String.Format(CultureInfo.CurrentCulture, ApplicationResources.ActivityNotificationSubject);
                 var emailBody = String.Format(CultureInfo.CurrentCulture, ApplicationResources.ActivityNotificationBody);
 
                 SendMessage(emailAddresses, emailSubject, emailBody);
             }
         }
+
+	    private static IList<string> getEmailAddresses(IList<Employee> peopleToEmail, IList<Employee> employees)
+	    {
+			IList<string> emailAddresses = new List<string>();
+			foreach (Employee person in peopleToEmail)
+			{
+				Debug.Assert(person.ApprovingManagerId1.HasValue || person.ApprovingManagerId2.HasValue);
```

Comment:
I think we might be using Log4Net in the rest of the app for debugging and instrumentation?